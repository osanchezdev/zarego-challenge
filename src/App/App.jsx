import React, { useState } from 'react';
import styles from './App.module.css';
import Popup from '../shared/Popup/Popup';
import ZaregoLogo from '../shared/ZaregoLogo/ZaregoLogo';

function App() {
  const [showPopup, setShowPopup] = useState(true);

  return (
    <main className={styles.app}>
      <div className={styles.app_logo_wrapper}>
        <ZaregoLogo handleClick={() => setShowPopup(true)} />
      </div>
      <Popup show={showPopup} handleClose={() => setShowPopup(false)}>
        <h1>Popup content</h1>
      </Popup>
    </main>
  );
}

export default App;
