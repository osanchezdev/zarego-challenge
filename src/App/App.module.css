.app {
  height: 100vh;
  display: grid;
  align-content: center;
  justify-content: center;
}

.app_logo__wrapper {
  text-align: center;
}
