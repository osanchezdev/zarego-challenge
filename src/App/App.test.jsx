import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

describe('App component', () => {
  beforeEach(() => {
    render(<App />);
  });

  it('should render zarego logo', () => {
    const logo = screen.getByAltText(/zarego/i);
    expect(logo).toBeInTheDocument();
  });

  it('should render popup', async () => {
    const popup = await screen.getByRole('dialog');
    expect(popup).toBeInTheDocument();
  });
});
