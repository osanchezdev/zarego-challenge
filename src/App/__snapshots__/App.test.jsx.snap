// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render component 1`] = `
<main
  className="app"
>
  <div
    className="app_logo_wrapper"
  >
    <div>
      <img
        alt="Zarego"
        className="zarego_logo"
        onClick={[Function]}
        src="logo.png"
      />
    </div>
  </div>
  <div
    className="popup__overlay popup__overlay_show"
    onClick={[Function]}
    role="dialog"
  >
    <div
      aria-hidden={true}
      className="popup__wrapper"
      onClick={[Function]}
    >
      <h1>
        Popup content
      </h1>
    </div>
  </div>
</main>
`;
